File: src/main/java/com/ftn/service/BusinessYearService.java
package com.ftn.service;

import com.ftn.model.BusinessYear;

import java.util.List;

/**
 * Created by JELENA on 30.5.2017.
 */
public interface BusinessYearService {

    List<BusinessYear> read();

    BusinessYear read(Long id);

    BusinessYear create(BusinessYear businessYear);

    BusinessYear update(Long id, BusinessYear businessYear);

    void delete(Long id);

    boolean active(BusinessYear businessYear);
}

File: src/main/webapp/service/businessYearService.js
/**
 * Created by JELENA on 3.6.2017.
 */

app.service('businessYearService', function ($http) {
    return {
        read: function (onSuccess, onError) {
            $http.get('/api/businessYears').then(onSuccess, onError);
        },
        create: function (businessYear, onSuccess, onError) {
            $http.post('/api/businessYears', businessYear).then(onSuccess, onError);
        },
        update: function (businessYear, onSuccess, onError) {
            $http.patch('/api/businessYears/' + businessYear.id, businessYear).then(onSuccess, onError);
        },
        delete: function (id, onSuccess, onError) {
            $http.delete('/api/businessYears/' + id).then(onSuccess, onError);
        }
    }
});